File: AlexanderKlimovLabs/3.Click/JavaVersion/Click/app/src/main/java/com/github/varenytsiamykhailo/alexanderklimov/click/MainActivity.java
package com.github.varenytsiamykhailo.alexanderklimov.click;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

import com.github.varenytsiamykhailo.alexanderklimov.click.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    private int mCounter = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //setContentView(R.layout.activity_main);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.buttonGreeting.setOnClickListener(view ->
                binding.textView.setText("Hello Kitty!"));

        binding.buttonCounter.setOnClickListener(view ->
                binding.textView.setText("Я насчитал " + ++mCounter + " ворон"));
    }
}

File: AlexanderKlimovLabs/3.Click/KotlinVersion/Click/app/src/main/java/com/github/varenytsiamykhailo/alexanderklimov/click/MainActivity.kt
package com.github.varenytsiamykhailo.alexanderklimov.click

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.github.varenytsiamykhailo.alexanderklimov.click.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    private var counter: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.buttonGreeting.setOnClickListener {
            binding.textView.text = "Hello Kitty!"
        }

        binding.buttonCounter.setOnClickListener {
            binding.textView.text = "я насчитал ${++counter} ворон"
        }
    }
}